scripts/school_and_ax/export_school_newsletter.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+"""Export the data needed for the school to send its newsletter to the subscribers
+
+Requires:
+    pip install PyMySQL
+    platal.conf file in the parent folder (shared with ../export_platal_to_json.py)
+"""
+import configparser
+import csv
+import datetime
+import os.path
+import re
+
+import pymysql
+
+
+# Load platal.conf and connect
+conf = configparser.SafeConfigParser()
+conf.read(os.path.join(os.path.dirname(__file__), '..', 'platal.conf'))
+db = pymysql.connect(
+    host=conf['Core']['dbhost'].strip('"'),
+    user=conf['Core']['dbuser'].strip('"'),
+    password=conf['Core']['dbpwd'].strip('"'),
+    db=conf['Core']['dbdb'].strip('"'),
+    charset='utf8mb4',
+)
+
+
+# Check that the school newsletter has the expected ID
+SCHOOL_NL_ID = 2
+with db.cursor() as cursor:
+    sql = """
+        SELECT  n.name, g.diminutif
+          FROM  newsletters AS n
+    INNER JOIN  groups AS g ON (g.id = n.group_id)
+         WHERE  n.id = %s
+    """
+    cursor.execute(sql, SCHOOL_NL_ID)
+    rows = cursor.fetchall()
+    if rows != (("Lettre de l'École polytechnique", "Ecole"), ):
+        raise RuntimeError("The school newsletter does not have ID {}: {}".format(SCHOOL_NL_ID, rows))
+
+
+# Create a CSV file
+csv_name = datetime.datetime.now().strftime('%Y-%m-%d_export_dixit.csv')
+with open(csv_name, 'w', newline='', encoding='utf-8') as fout:
+    csv_stream = csv.writer(fout, delimiter=';', quotechar='"', escapechar='\\', quoting=csv.QUOTE_MINIMAL)
+    csv_stream.writerow(("Prénom", "Nom", "Promotion", "Adresse"))
+
+    # Export subscribers
+    with db.cursor() as cursor:
+        sql = """
+           SELECT  a.firstname, a.lastname, pd.promo, CONCAT(s.email, '_ecole@', d.name) AS email
+             FROM  newsletter_ins AS ni
+        LEFT JOIN  accounts AS a ON (a.uid = ni.uid)
+        LEFT JOIN  account_profiles AS ap ON (ap.uid = a.uid AND FIND_IN_SET('owner', ap.perms))
+        LEFT JOIN  profiles AS p ON (p.pid = ap.pid)
+        LEFT JOIN  profile_display AS pd ON (pd.pid = p.pid)
+        LEFT JOIN  email_source_account AS s ON (s.uid = a.uid AND s.type = 'forlife')
+        LEFT JOIN  email_virtual_domains AS m ON (s.domain = m.id)
+        LEFT JOIN  email_virtual_domains AS d ON (d.aliasing = m.id)
+            WHERE  ni.nlid = %s
+                   AND p.deathdate IS NULL
+                   AND pd.promo IS NOT NULL
+                   AND d.name = "alumni.polytechnique.org"
+         GROUP BY  a.uid
+         ORDER BY  pd.promo, pd.sort_name, a.uid
+        """
+        cursor.execute(sql, SCHOOL_NL_ID)
+        for row in cursor:
+            # Sanity checks before adding a row, in order to make sure we do not transmit invalid data to the school
+            firstname, lastname, promo, email = row
+
+            name_charset = r"[^'.a-zA-ZÁÇÉÖàáâãäåçèéêëíîïñóôöúüÿ -]"
```

Comment:
Why this set of valid chars?

Reply:
Because it is the charset of all the names currently in the database. If you know a reference set of characters that are acceptable in names, it can be changed to it.